### PR TITLE
fix(catering): remove ability to edit existing orders when field disabled (PPT-1340) (PPT-1339)

### PR DESCRIPTION
**Pull request by MrYuion**

### CODE REVIEW

**Comment by w-le** [comment]
Thanks, once done remove Draft and I'll get @KesterJJ to test

**Comment by w-le** [comment]
Hey @ksra08  can I get a test of this one? Thanks

**Comment by w-le** [comment]
Hey @KesterJJ can I get a test of this one? Deploy this build branch to bm-stg workplace-pos, Thanks

**Comment by KesterJJ** [comment]
Tested working

File: apps/workplace/src/app/book/meeting-flow/meeting-flow-form.component.ts [hunk 1]
```diff
@@ -464,6 +472,117 @@ export class MeetingFlowFormComponent extends AsyncHandler {
...
+                        (s) =>
+                            items.filter(
+                                (_) =>
+                                    !(_ as any).hide_for_zones?.find((z) =>
+                                        s.zones.includes(z)
+                                    )
+                            ).length > 0
+                    );
+                    if (
+                        can_cater &&
```
Comment (w-le) [inline apps/workplace/src/app/book/meeting-flow/meeting-flow-form.component.ts:505]: Hey @MrYuion is this catering var relevant inside this assets function? Or is it just a leftover from copy/paste that shouldn't be here?

File: apps/workplace/src/app/book/meeting-flow/meeting-flow-form.component.ts [hunk 2]
```diff
@@ -464,6 +472,117 @@ export class MeetingFlowFormComponent extends AsyncHandler {
...
+                            space_list.find((i) => i.id === _)
+                        )
+                    )
+                        return true;
+                    const event = this._state.event;
+                    const { id, catering, date, date_end } =
+                        this.form.getRawValue();
+                    const time_changed =
+                        !id ||
+                        (catering?.length &&
```
Comment (w-le) [inline apps/workplace/src/app/book/meeting-flow/meeting-flow-form.component.ts:516]: Hey @MrYuion is this catering var relevant inside this assets function? Or is it just a leftover from copy/paste that shouldn't be here?

**Pull request merged**